docs/user-guide/async.rst

```diff
@@ -0,0 +1,36 @@
+
+Async Zarr
+==========
+
+Zarr-Python 3 added a new asynchronous API for reading and writing data using `asyncio`.
```

Comment:
Would be nice to add a quick motivation here - what benefits does the async API provide?